packages/flutter/lib/src/material/reorderable_list.dart

```diff
@@ -0,0 +1,402 @@
+/// by [ReorderableListView] will need to account for this when inserting before
+/// [newIndex].
+/// 
+/// Example implementation:
+/// 
+/// ```dart
+/// final List<MyDataObject> backingList = <MyDataObject>[/* ... */];
+/// void onSwap(int oldIndex, int newIndex) {
+///   if (oldIndex < newIndex) {
+///     // removing the item at oldIndex will shorten the list by 1.
+///     newIndex -= 1;
+///   }
+///   final MyDataObject element = backingList.removeAt(oldIndex);
+///   backingList.insert(newIndex, element);
+/// }
+/// ```
+typedef void OnSwapCallback(int oldIndex, int newIndex);
+
+/// A list with draggable content that the user can re-order.
+/// 
+/// Note that this widget places its [children] in a [Column] or [Row] and not a [ListView].
+///
+/// All [children] must have a key.
+class ReorderableListView extends StatefulWidget {
+
+  /// Creates a reorderable list.
+  const ReorderableListView({
+    @required this.children, 
+    @required this.onSwap, 
+    this.scrollDirection = Axis.vertical, 
+    this.padding, 
+    this.dropAreaExtent = 96.0,
+  }): assert(dropAreaExtent != null),
+      assert(scrollDirection != null),
+      assert(onSwap != null),
+      assert(children != null);
+
+  /// The widgets to display.
+  final List<Widget> children;
+
+  /// The [Axis] along which the list scrolls.
+  /// 
+  /// List children also drag along this [Axis].
+  final Axis scrollDirection;
+
+  /// The amount of space by which to inset the child.
+  final EdgeInsets padding;
+
+  /// Called when a list child is dropped into a new position to shuffle the
+  /// underlying list.
+  /// 
+  /// This [ReorderableListView] calls [onSwap] after a list child is dropped
+  /// into a new position.
+  final OnSwapCallback onSwap;
+
+  /// The extent along the [scrollDirection] axis to allow a child to drop
+  /// into when the user reorders list children.
+  /// 
+  /// When the user is dragging a child to reorder it, the list will open up a
+  /// drop area in the space under the widget being dragged. That area will
+  /// have an extent along the main axis of [dropAreaExtent].
+  /// 
+  /// In a vertical list, this should match the height of the list items.
+  /// In a horizontal list, this should match the width of the list items.
+  final double dropAreaExtent;
+
+  @override
+  State<StatefulWidget> createState() {
+    return new _ReorderableListViewState();
+  }
+}
+
+class _ReorderableListViewState extends State<ReorderableListView> {
+  // We use an inner overlay so that the dragging list item doesn't draw outside of the list itself.
+  GlobalKey _overlayKey;
+
+  // This entry contains the scrolling list itself.
+  OverlayEntry _bottomOverlayEntry;
+
+  @override 
+  void initState() {
+    super.initState();
+    _overlayKey = new GlobalKey(debugLabel: '$this overlay key');
+    _bottomOverlayEntry = new OverlayEntry(
+      opaque: true,
+      builder: _buildOverlayContent,
```

Comment:
Why not just `builder: (BuildContext context) => new _ReorderableListContent(widget)`